src/Passes/Source/ModuleLoader/ModuleLoader.hpp

```diff
@@ -0,0 +1,132 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "RemoveDisallowedAttribubtesPass/RemoveDisallowedAttribubtesPass.hpp"
+#include "Types/Types.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    class SingleModuleTransformation
```

Comment:
I see that this class is used inside of a single member function only.
Consider nesting the class as deep as reasonable.

If the class is not expected to be used outside of that member function then consider defining this class inside of that member function.
If the class is not expected to be used outside of `class ModuleLoader` then consider defining this class inside of a non-public section of `class ModuleLoader`.
If the class can be used outside of `class ModuleLoader` but can only exist together with `class ModuleLoader` (i.e. if `class ModuleLoader` is ever removed then the `class SingleModuleTransformation` must also be removed) then consider defining this class inside of public section of `class ModuleLoader`.

Reply:
Done.